pkg/controller/disruption/disruption.go

```diff
@@ -476,33 +486,72 @@ func (dc *DisruptionController) removeDb(obj interface{}) {
 	dc.enqueuePdb(pdb)
 }
 
-func (dc *DisruptionController) addPod(obj interface{}) {
-	pod := obj.(*v1.Pod)
-	klog.V(4).Infof("addPod called on pod %q", pod.Name)
-	pdb := dc.getPdbForPod(pod)
-	if pdb == nil {
-		klog.V(4).Infof("No matching pdb for pod %q", pod.Name)
-	} else {
-		klog.V(4).Infof("addPod %q -> PDB %q", pod.Name, pdb.Name)
-		dc.enqueuePdb(pdb)
-	}
+type podLabelItem struct {
+	encodedLabels string
+	namespace     string
+}
+
+func (dc *DisruptionController) enqueuePod(pod *v1.Pod) {
```

Comment:
logically, it looks like the only thing we care about here are the labels.

I wonder if three queue levels would make more sense:
1. process pods
  a. add their labels to a map[namespace]{podName, podNamespace, labels}
  b. queue their namespace for processing
2. process namespaces
  a. grab all the sets of labels observed for that namespace
  b. resolve to covering pdbs (logging using the podName/podNamespace if there are multiple pdbs found or no pdbs found)
  c. queue the pdbs for processing
3. process pdbs

as an aside, processing pod-by-pod assumes we will see every pod event... on startup, it seems like we have to do an initial update of all pdbs to cover any pod events we missed... are we doing that?

Reply:
>I wonder if three queue levels would make more sense
I'm not sure I understand what's the benefit of the suggested approach.

>it seems like we have to do an initial update of all pdbs to cover any pod events we missed... are we doing that?
More or less.
On startup we enqueue all pdbs to DisruptionController.queue, on top of that we also enqueue all unique pod labels in labelUpdateQueue and eventually converge all pdbs which match those labels.